### PR TITLE
Improved About app 

### PR DESCRIPTION
**Pull request by Agrim1234**
contributors changed and made like ard
github, linkedin and website link also added
more by mac section made

### CODE REVIEW

File: androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java [hunk 1]
```diff
@@ -27,22 +27,39 @@
 
 public class AboutUsActivity extends AppCompatActivity implements View.OnClickListener {
 
+    private static final String TAG = "mac";
     public static String ABOUT_US_FACEBOOK_URL = "https://www.facebook.com/MACBITSGoa";
     public static String ABOUT_US_FACEBOOK_PAGE_ID = "MACBITSGoa";
+    public static String ABOUT_US_GITHUB_URL = "https://github.com/MobileApplicationsClub";
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java]: Take these URLs from firebase

File: androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java [hunk 2]
```diff
@@ -27,22 +27,39 @@
...
+        ImageButton linkedinImgBtn = findViewById(R.id.content_about_us_linkedin_imgbtn);
+        ImageButton websiteImgBtn = findViewById(R.id.content_about_us_website_imgbtn);
+
+        android.widget.ListView morebymacListView = findViewById(R.id.content_about_us_morebymac_lv);
+        androidx.recyclerview.widget.RecyclerView contributorsRecyclerView = findViewById(R.id.content_about_us_contributors_rv);
 
         setSupportActionBar(toolbar);
         getSupportActionBar().setDisplayHomeAsUpEnabled(true);
-        getSupportActionBar().setTitle("About DoJMA Android App");
+        getSupportActionBar().setTitle("About App");
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java]: Use String res

File: androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java [hunk 3]
```diff
@@ -59,21 +76,47 @@ protected void onCreate(final Bundle savedInstanceState) {
             window.setNavigationBarColor(ContextCompat.getColor(this, R.color.mac_color));
         }
 
-        List<String> contributors = generateContributorsData();
+        final String[] str2 = new String[]{"ARD App","NMD App","Bits Login App","Abhigyan App","ICEF App"};
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java]: From firebase, no hard coding

File: androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java [hunk 4]
```diff
@@ -59,21 +76,47 @@ protected void onCreate(final Bundle savedInstanceState) {
...
+        android.widget.ArrayAdapter<String> adapter = new android.widget.ArrayAdapter<>(context, android.R.layout.simple_list_item_1, str2);
+        morebymacListView.setAdapter(adapter);
+
+        morebymacListView.setOnItemClickListener(new android.widget.AdapterView.OnItemClickListener() {
+
+            @Override
+            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
+                switch (i)
+                {
+                    case 0 :Intent ardIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(ABOUT_US_ARD_URL));
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java]: Use chrome custom tabs

File: androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java [hunk 5]
```diff
@@ -92,9 +135,24 @@ public void onClick(View v) {
                     startActivity(intent);
                 }
                 break;
+            case R.id.content_about_us_linkedin_imgbtn:
+                Intent linkedinIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(ABOUT_US_LINKEDIN_URL));
+                Toast.makeText(this, "Opening in browser", Toast.LENGTH_SHORT).show();
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java]: Remove toast

File: androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java [hunk 6]
```diff
@@ -92,9 +135,24 @@ public void onClick(View v) {
...
+                break;
+            case R.id.content_about_us_github_imgbtn:
+                Intent githubIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(ABOUT_US_GITHUB_URL));
+                Toast.makeText(this, "Opening in browser", Toast.LENGTH_SHORT).show();
+                startActivity(githubIntent);
+                break;
             case R.id.content_about_us_google_play_imgbtn:
                 Intent googlePlayIntent = new Intent(Intent.ACTION_VIEW);
-                googlePlayIntent.setData(Uri.parse("market://search?q=pub:Mobile App Club - BITS Goa"));
+                googlePlayIntent.setData(Uri.parse("market://search?q=Mobile App Club - BITS Goa"));
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java]: Use exact URL

File: androidClient/app/src/main/java/com/csatimes/dojma/adapters/ContributorsAdapter.java [hunk 7]
```diff
@@ -13,26 +13,89 @@
...
     @Override
     public int getItemCount() {
-        return contributorsList.size();
+        return (contributors.size());
+    }
+
+    private void populate() {
+        contributors = new ArrayList<>();
+        contributors.add(new Contributor(
+                "Vikramaditya Kukreja",
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/adapters/ContributorsAdapter.java]: Please, no hard coding

File: androidClient/app/src/main/java/com/csatimes/dojma/adapters/ContributorsAdapter.java [hunk 8]
```diff
@@ -13,26 +13,89 @@
...
+    private Context context;
+    private Activity activity;
+    Exception e;
 
-    public ContributorsAdapter(List<String> contributorsList) {
-        this.contributorsList = contributorsList;
+    public ContributorsAdapter(Context context) {
+        this.context = context;
+        populate();
+        DatabaseReference devRef = FirebaseDatabase.getInstance().getReference()
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/adapters/ContributorsAdapter.java]: Put this in populate, remove hardcoded values in `populate()`

File: androidClient/app/src/main/java/com/csatimes/dojma/viewholders/ContributorsViewHolder.java [hunk 9]
```diff
@@ -10,11 +10,37 @@
...
+    public TextView nameTv;
+    public TextView emailTv;
+    public TextView phoneTv;
+    public SimpleDraweeView photoIv;
+    public Activity activity;
+    public Contributor contributor;
+    public Context context ;
 
-    public ContributorsViewHolder(View itemView) {
+    public ContributorsViewHolder(View itemView, final Context context) {
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/viewholders/ContributorsViewHolder.java]: Don't pass context by parameter, use `itemView.getContext()`

File: androidClient/app/src/main/res/drawable/ic_linkedin1.xml [hunk 10]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (Rushi98) [inline androidClient/app/src/main/res/drawable/ic_linkedin1.xml]: Why LinkedIn**1** ? (Filename)

File: androidClient/app/src/main/res/layout/content_about_us.xml [hunk 11]
```diff
@@ -58,30 +57,73 @@
         android:id="@+id/content_about_us_fb_imgbtn"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:layout_margin="4dp"
-        android:layout_marginTop="8dp"
+        android:layout_marginLeft="112dp"
```
Comment (Rushi98) [inline androidClient/app/src/main/res/layout/content_about_us.xml]: 112 dp why?

File: androidClient/app/src/main/res/layout/content_about_us.xml [hunk 12]
```diff
@@ -58,30 +57,73 @@
...
         app:layout_constraintTop_toBottomOf="@+id/content_about_us_mac_img"
         app:srcCompat="@drawable/ic_facebook_white" />
 
     <ImageButton
         android:id="@+id/content_about_us_google_play_imgbtn"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:layout_margin="4dp"
-        android:layout_marginTop="8dp"
+        android:layout_marginRight="210dp"
```
Comment (Rushi98) [inline androidClient/app/src/main/res/layout/content_about_us.xml]: Why 210 dp?

File: androidClient/app/src/main/res/layout/content_about_us.xml [hunk 13]
```diff
@@ -58,30 +57,73 @@
...
+    <ImageButton
+        android:id="@+id/content_about_us_github_imgbtn"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="4dp"
+        android:layout_marginLeft="80dp"
+        android:background="@color/mac_color"
+        android:textColor="@color/white"
+        app:srcCompat="@drawable/ic_github_circle"
+        app:layout_constraintHorizontal_chainStyle="packed"
```
Comment (Rushi98) [inline androidClient/app/src/main/res/layout/content_about_us.xml]: What does this do?

File: androidClient/app/src/main/res/layout/content_about_us.xml [hunk 14]
```diff
@@ -98,6 +140,33 @@
...
+        android:id="@+id/content_about_us_morebymac_tv"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginLeft="16dp"
+        android:layout_marginStart="16dp"
+        android:layout_marginTop="16dp"
+        android:gravity="center"
+        android:padding="8dp"
+        android:text="More By MAC"
+        app:layout_constraintLeft_toLeftOf="parent"
```
Comment (Rushi98) [inline androidClient/app/src/main/res/layout/content_about_us.xml]: String resource

File: androidClient/app/src/main/res/layout/content_about_us.xml [hunk 15]
```diff
@@ -98,6 +140,33 @@
...
+        app:layout_constraintTop_toBottomOf="@+id/content_about_us_desc_tv" />
+
+    <android.widget.ListView
```
Comment (Rushi98) [inline androidClient/app/src/main/res/layout/content_about_us.xml]: Use recycler view

File: androidClient/app/src/main/res/layout/content_about_us.xml [hunk 16]
```diff
@@ -98,6 +140,33 @@
...
+        android:id="@+id/content_about_us_morebymac_lv"
+        android:layout_width="0dp"
+        android:layout_height="250dp"
```
Comment (Rushi98) [inline androidClient/app/src/main/res/layout/content_about_us.xml]: Why fixed height?

**Review by Rushi98 — CHANGES_REQUESTED** [review]
Remove PNG images, use vector drawables

File: androidClient/app/src/main/res/layout/content_about_us.xml (hunk 15 above)
Comment (Agrim1234) [reply androidClient/app/src/main/res/layout/content_about_us.xml]: tanmay said to use list  view

File: androidClient/app/src/main/res/layout/content_about_us.xml (hunk 13 above)
Comment (Agrim1234) [reply androidClient/app/src/main/res/layout/content_about_us.xml]: this is github icon

File: androidClient/app/src/main/java/com/csatimes/dojma/adapters/ContributorsAdapter.java (hunk 7 above)
Comment (Agrim1234) [reply androidClient/app/src/main/java/com/csatimes/dojma/adapters/ContributorsAdapter.java]: this is for the case when there is no internet connection. It is same as did in ard and as was said by tanmay

File: androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java (hunk 2 above)
Comment (Agrim1234) [reply androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java]: ok

File: androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java (hunk 5 above)
Comment (Agrim1234) [reply androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java]: ok

File: androidClient/app/src/main/java/com/csatimes/dojma/adapters/ContributorsAdapter.java (hunk 7 above)
Comment (Rushi98) [reply androidClient/app/src/main/java/com/csatimes/dojma/adapters/ContributorsAdapter.java]: we will enable firebase persistence, so offline will not be issue

File: androidClient/app/src/main/res/layout/content_about_us.xml (hunk 13 above)
Comment (Rushi98) [reply androidClient/app/src/main/res/layout/content_about_us.xml]: the `app:layout_constraintHorizontal_chainStyle="packed"` thing

File: androidClient/app/src/main/res/layout/content_about_us.xml (hunk 15 above)
Comment (Rushi98) [reply androidClient/app/src/main/res/layout/content_about_us.xml]: @tanmaydixit we should use recyclerview for future customizations

Comment (tanmaydixit) [reply androidClient/app/src/main/res/layout/content_about_us.xml]: Yes i didn't think we would need to further customize, but yes a recyclerview would be better.
@Agrim1234 My bad.

Comment (Agrim1234) [reply androidClient/app/src/main/res/layout/content_about_us.xml]: ok i will do it

File: androidClient/app/src/main/res/layout/content_about_us.xml (hunk 16 above)
Comment (Agrim1234) [reply androidClient/app/src/main/res/layout/content_about_us.xml]: wrap content was not working properly

File: androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java (hunk 2 above)
Comment (Agrim1234) [reply androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java]: done

Comment (Rushi98) [reply androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java]: commit and push the changes

File: androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java (hunk 5 above)
Comment (Agrim1234) [reply androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java]: done

File: androidClient/app/src/main/res/layout/content_about_us.xml (hunk 16 above)
Comment (Rushi98) [reply androidClient/app/src/main/res/layout/content_about_us.xml]: as all members are one below other, you can use linearlayout. (maybe put it inside nested scroll view https://developer.android.com/reference/android/support/v4/widget/NestedScrollView )
So finally something like
```xml
<nestedscrollview>
  <linearlayout>
    <some... views/>
  </linearlayout>
</nestedscrollview>

File: androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java (hunk 6 above)
Comment (Agrim1234) [reply androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java]: done

File: androidClient/app/src/main/java/com/csatimes/dojma/adapters/ContributorsAdapter.java [hunk 17]
```diff
@@ -1,38 +1,113 @@
...
+        contributors.clear();
+        for (DataSnapshot child :
+                dataSnapshot.getChildren()) {
+            contributors.add(child.getValue(Contributor.class));
+        }
+        notifyDataSetChanged();
+    }
+
+    @Override
+    public void onCancelled(DatabaseError databaseError) {
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/adapters/ContributorsAdapter.java]:
```java
Log.e(TAG, databaseError.getMessage(), databaseError.toException());
```

File: androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java [hunk 18]
```diff
@@ -27,22 +29,40 @@
 
 public class AboutUsActivity extends AppCompatActivity implements View.OnClickListener {
 
+    private static final String TAG = "mac";
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java]:
```java
TAG = TAG_PREFIX + AboutUsActivity.class.getSimpleName();
```
Delete if unused.

File: androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java [hunk 19]
```diff
@@ -27,22 +29,40 @@
...
+    public static String ABOUT_US_WEBSITE_URL = "https://macbitsgoa.com";
+    public static String ABOUT_US_ARD_URL = "https://play.google.com/store/apps/details?id=com.macbitsgoa.ard";
+    public static String ABOUT_US_NMD_URL = "https://play.google.com/store/apps/details?id=com.macbitsgoa.nmd";
+    public static String ABOUT_US_BL_URL = "https://play.google.com/store/apps/details?id=net.deepeshmakhijani.bpgclogin";
+    public static String ABOUT_US_ABHIGYAN_URL = "https://play.google.com/store/apps/details?id=com.macbitsgoa.abhigyaan";
+    public static String ABOUT_US_ICEF_URL = "https://play.google.com/store/apps/details?id=bits.mac.icef_2018";
+    public static String ABOUT_US_GOOGLEPLAY_URL = "https://play.google.com/store/search?q=Mobile%20App%20Club%20-%20BITS%20Goa&c=apps&hl=en";
+    Context context = AboutUsActivity.this;
+    Activity activity = AboutUsActivity.this;
+    Exception e;
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java]: What's use of `context`, `activity` and `e`?

File: androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java [hunk 20]
```diff
@@ -27,22 +29,40 @@
...
-        androidx.recyclerview.widget.RecyclerView contributorsRecyclerView = (RecyclerView) findViewById(R.id.content_about_us_rv);
+        Toolbar toolbar = findViewById(R.id.about_us_toolbar);
+        ImageButton facebookImgBtn = findViewById(R.id.content_about_us_fb_imgbtn);
+        ImageButton googlePlayImgBtn = findViewById(R.id.content_about_us_google_play_imgbtn);
+        ImageButton githubImgBtn = findViewById(R.id.content_about_us_github_imgbtn);
+        ImageButton linkedinImgBtn = findViewById(R.id.content_about_us_linkedin_imgbtn);
+        ImageButton websiteImgBtn = findViewById(R.id.content_about_us_website_imgbtn);
+
+        android.widget.ListView morebymacListView = findViewById(R.id.content_about_us_morebymac_lv);
+        androidx.recyclerview.widget.RecyclerView contributorsRecyclerView = findViewById(R.id.content_about_us_contributors_rv);
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java]: remove unnecessary qualifier

File: androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java [hunk 21]
```diff
@@ -59,21 +79,47 @@ protected void onCreate(final Bundle savedInstanceState) {
...
-        List<String> contributors = generateContributorsData();
+        final String[] str2 = new String[]{"ARD App","NMD App","Bits Login App","Abhigyan App","ICEF App"};
+        android.widget.ArrayAdapter<String> adapter = new android.widget.ArrayAdapter<>(context, android.R.layout.simple_list_item_1, str2);
+        morebymacListView.setAdapter(adapter);
+
+        morebymacListView.setOnItemClickListener(new android.widget.AdapterView.OnItemClickListener() {
+
+            @Override
+            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
+                switch (i)
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java]: add a default clause

File: androidClient/app/build.gradle [hunk 22]
```diff
@@ -61,6 +61,8 @@ dependencies {
 
     implementation "com.facebook.fresco:fresco:1.9.0"
 
+    //noinspection GradleCompatible
+    implementation 'com.android.support:customtabs:23.3.0'
```
Comment (Rushi98) [inline androidClient/app/build.gradle]: https://developer.android.com/topic/libraries/support-library/refactor
`androidx.browser:browser:1.0.0-alpha1`

**Comment by Rushi98** [comment]
@Agrim1234 fix codacy  issues

File: androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java [hunk 23]
```diff
@@ -29,20 +32,44 @@
 
     public static String ABOUT_US_FACEBOOK_URL = "https://www.facebook.com/MACBITSGoa";
     public static String ABOUT_US_FACEBOOK_PAGE_ID = "MACBITSGoa";
+    public static String ABOUT_US_GITHUB_URL = "https://github.com/MobileApplicationsClub";
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java]: I meant all of the urls to be taken from firebase. Also, take urls for the icons from firebase.

File: androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java [hunk 24]
```diff
@@ -29,20 +32,44 @@
...
     @Override
     protected void onCreate(final Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_about_us);
 
-        Toolbar toolbar = (Toolbar) findViewById(R.id.about_us_toolbar);
-        ImageButton facebookImgBtn = (ImageButton) findViewById(R.id.content_about_us_fb_imgbtn);
-        ImageButton googlePlayImgBtn = (ImageButton) findViewById(R.id.content_about_us_google_play_imgbtn);
-        androidx.recyclerview.widget.RecyclerView contributorsRecyclerView = (RecyclerView) findViewById(R.id.content_about_us_rv);
+        Toolbar toolbar = findViewById(R.id.about_us_toolbar);
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/activities/AboutUsActivity.java:26]: keep all links in a single rv, each item will have icon url, title, and link to open on click. All of these from firebase

File: androidClient/app/src/main/java/com/csatimes/dojma/adapters/ContributorsAdapter.java [hunk 25]
```diff
@@ -1,38 +1,77 @@
...
     @Override
     public void onBindViewHolder(ContributorsViewHolder holder, int position) {
-        holder.nameTextView.setText(contributorsList.get(position));
+        holder.populate(contributors.get(position));
     }
 
     @Override
     public int getItemCount() {
-        return contributorsList.size();
+        if(contributors == null)
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/adapters/ContributorsAdapter.java:52]: curly brackets

File: androidClient/app/src/main/java/com/csatimes/dojma/adapters/ContributorsAdapter.java [hunk 26]
```diff
@@ -1,38 +1,77 @@
...
+            return 0;
+        else
+            return contributors.size();
+    }
+
+    @Override
+    public void onDataChange(DataSnapshot dataSnapshot) {
+        if (dataSnapshot == null)
+            return;
+        if (contributors != null)
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/adapters/ContributorsAdapter.java:62]: where are contributors initialised?

File: androidClient/app/src/main/java/com/csatimes/dojma/adapters/MorebymacAdapter.java [hunk 27]
```diff
@@ -0,0 +1,74 @@
...
+            return moreApps.size();
+    }
+
+    @Override
+    public void onDataChange(DataSnapshot dataSnapshot) {
+        if (dataSnapshot == null)
+            return;
+
+        if (moreApps != null)
+        {moreApps.clear();}
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/adapters/MorebymacAdapter.java]: indent properly

File: androidClient/app/src/main/java/com/csatimes/dojma/adapters/MorebymacAdapter.java [hunk 28]
```diff
@@ -0,0 +1,74 @@
...
+
+import java.util.ArrayList;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+import static androidx.constraintlayout.widget.Constraints.TAG;
+
+public class MorebymacAdapter extends RecyclerView.Adapter<MorebymacViewHolder> implements ValueEventListener{
+
+    private ArrayList<Morebymac> moreApps;
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/adapters/MorebymacAdapter.java]: where is this initialised?

File: androidClient/app/src/main/java/com/csatimes/dojma/models/Morebymac.java [hunk 29]
```diff
@@ -0,0 +1,15 @@
+package com.csatimes.dojma.models;
+
+public class Morebymac {
+    public String name;
+    public String web;
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/models/Morebymac.java]: Do you mean `appName` and `playStoreUrl`?

File: androidClient/app/src/main/java/com/csatimes/dojma/utilities/DHC.java [hunk 30]
```diff
@@ -88,6 +83,16 @@
      */
     public static final String UPDATE_SERVICE_HERALD_PAGES = "HERALD_PAGES";
 
+    /**
+     * Firebase node name for moreapps.
+     */
+    public static final String FIREBASE_DATABASE_REFERENCE_MOREAPPS = "moreapps";
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/utilities/DHC.java]: keep all firebase node names in separate class `FirebaseKeys` similar to `SpKeys`. (For now you can add only those you've created)

File: androidClient/app/src/main/java/com/csatimes/dojma/viewholders/ContributorsViewHolder.java [hunk 31]
```diff
@@ -1,20 +1,55 @@
...
  */
 
-public class ContributorsViewHolder extends RecyclerView.ViewHolder {
-    public TextView nameTextView;
+public class ContributorsViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+    public TextView nameTv;
+    public TextView emailTv;
+    public TextView phoneTv;
+    public SimpleDraweeView photoIv;
+    public Activity activity;
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/viewholders/ContributorsViewHolder.java]: where is `activity` used?

File: androidClient/app/src/main/java/com/csatimes/dojma/viewholders/ContributorsViewHolder.java [hunk 32]
```diff
@@ -1,20 +1,55 @@
...
+        nameTv.setText(contributor.name);
+        emailTv.setText(contributor.email);
+        phoneTv.setText(contributor.phone);
+        photoIv.setImageURI(Uri.parse(contributor.photoUrl));
+    }
+
+    @Override
+    public void onClick(View view) {
+        Intent websiteIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(contributor.web));
+        context.startActivity(websiteIntent);
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/viewholders/ContributorsViewHolder.java]: remove global variable `context`, use `view.getContext`

File: androidClient/app/src/main/java/com/csatimes/dojma/viewholders/ContributorsViewHolder.java [hunk 33]
```diff
@@ -1,20 +1,55 @@
...
+        this.contributor = contributor;
+        nameTv.setText(contributor.name);
+        emailTv.setText(contributor.email);
+        phoneTv.setText(contributor.phone);
+        photoIv.setImageURI(Uri.parse(contributor.photoUrl));
+    }
+
+    @Override
+    public void onClick(View view) {
+        Intent websiteIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(contributor.web));
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/viewholders/ContributorsViewHolder.java]: launch `customTab` (You may need `activity` for it, then you can keep it)

File: androidClient/app/src/main/java/com/csatimes/dojma/viewholders/MorebymacViewHolder.java [hunk 34]
```diff
@@ -0,0 +1,65 @@
...
+    public Activity activity;
+    public Morebymac morebymac;
+    public Context context ;
+
+    private Intent copy_intent = new Intent(context, CopyLinkBroadcastReceiver.class);
+    private PendingIntent copy_pendingIntent = PendingIntent.getBroadcast(context, 0, copy_intent, PendingIntent.FLAG_UPDATE_CURRENT);
+    private String copy_label = "Copy Link";
+
+    private int colorResource = getChromeCustomTabColorFromTheme();
+    private final CustomTabsIntent customTabsIntent = new CustomTabsIntent.Builder()
```
Comment (Rushi98) [inline androidClient/app/src/main/java/com/csatimes/dojma/viewholders/MorebymacViewHolder.java]: Add and use https://raw.githubusercontent.com/MobileApplicationsClub/kwiki/master/Chrome%20Custom%20Tabs/Browser.java for custom tabs

File: androidClient/app/src/main/res/drawable/ic_github_circle.xml [hunk 35]
```diff
@@ -0,0 +1,10 @@
+<!-- drawable/github_circle.xml -->
```
Comment (Rushi98) [inline androidClient/app/src/main/res/drawable/ic_github_circle.xml]: can delete these icons if taking image Urls from firebase

**Review by Rushi98 — CHANGES_REQUESTED** [review]
take social links from firebase

**Pull request closed without merging**